drivers/frequency/hmc7044/hmc7044.c

```diff
@@ -640,6 +645,122 @@ static int32_t hmc7044_setup(struct hmc7044_dev *dev)
 	return SUCCESS;
 }
 
+/**
+ * Setup the device.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t hmc7043_setup(struct hmc7044_dev *dev)
+{
+	struct hmc7044_chan_spec *chan;
+	uint32_t i;
+
+	if (dev->clkin_freq_ccf[0])
+		dev->pll2_freq = dev->clkin_freq_ccf[0];
+	else
+		dev->pll2_freq  = dev->clkin_freq[0];
+
+	if (!dev->pll2_freq) {
+		printf("%s: Failed to get valid parent rate\n", __func__);
```

Comment:
Is it ok to have printf in drivers?

Reply:
Usualy it helps to get some feedback in the setup phase.